src/net.c

```diff
@@ -457,6 +472,39 @@ coap_new_context(
 
   memset(c, 0, sizeof(coap_context_t));
 
+#ifdef HAVE_SYS_EPOLL_H
+  c->epfd = epoll_create1(0);
+  if (c->epfd == -1) {
+    coap_log(LOG_ERR, "coap_new_context: Unable to epoll_create: %s (%d)\n",
+             coap_socket_strerror(),
+             errno);
+  }
+#ifdef HAVE_SYS_TIMERFD_H
+  c->eptimerfd = timerfd_create(CLOCK_REALTIME, TFD_NONBLOCK);
+  if (c->eptimerfd == -1) {
+    coap_log(LOG_ERR, "coap_new_context: Unable to timerfd_create: %s (%d)\n",
+             coap_socket_strerror(),
+             errno);
+  }
+  else {
+    int ret;
+    struct epoll_event event;
+
+    event.events = EPOLLIN;
+    /* We special case this event by setting to NULL */
+    event.data.ptr = NULL;
+
+    ret = epoll_ctl(c->epfd, EPOLL_CTL_ADD, c->eptimerfd, &event);
```

Comment:
You did not check if the epfd is valid

Reply:
Thanks for finding this.  Updated code now pushed.